src/include/platform/ConfigurationManager.h

```diff
@@ -140,6 +140,13 @@ class ConfigurationManager
     virtual bool IsFullyProvisioned()   = 0;
     virtual void InitiateFactoryReset() = 0;
 
+    // Gets called when starting BLE/DNS-SD advertisement
```

Comment:
@hnnajh I did some testing on nrfconnect platform and I can confirm the counter value is changing, however the behavior looks like this:
- After boot and starting advertising (opening commissioning window) the counter value is equal to 3
- After advertising timeout and starting it again (open commissioning window) the counter value is equal to 5
- Subsequent attempts increment counter by 2 every time
I think that with current implementation you increment counter on every BLE advertising start and on every commissionable node discovery start independently, so it is incremented once on after boot (init starts commissionable discovery) and then two times on every open commissioning window. Is that behavior expected?

I don't have strong opinion on that, but I would expect the counter will be incremented once per every open commissioning window and the counter value will be kept consistent between BLE and DNS-SD advertisements. If you would agree with me on that please consider putting the following code:

> #if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
    // Increment life time counter to protect against long-term tracking of rotating device ID.
    ConfigurationMgr().IncrementLifetimeCounter();
#endif

in this method: https://github.com/project-chip/connectedhomeip/blob/master/src/app/server/CommissioningWindowManager.cpp#L271. That should increment counter on every commissioning window open in a common generic way for both DNS-SD and BLE without a need of splitting the code on the two cases.

Reply:
Agree, we should avoid incrementing 2 times and this seems like a reasonable fix

Reply:
Fixed